packages/sprinkles/src/createSprinkles.ts

```diff
@@ -60,16 +60,35 @@ type SprinkleProps<Args extends ReadonlyArray<any>> = Args extends [
       SprinkleProps<R>
   : {};
 
+type SprinkleConditions<Args extends ReadonlyArray<any>> = Args extends [
+  infer L,
+  ...infer R
+]
+  ? L extends SprinklesProperties
+    ? [
+        keyof L['conditions'],
+        ...SprinkleConditions<[ChildSprinkleProps<L['styles']>]>,
```

Comment:
Not sure why we need to iterate over the `ChildSprinkleProps`? Think this line isn't required. However I could be wrong as I often forgot the underlying data structures in here.

Reply:
you are absolutely right there is no need for that, my bad